crates/pg_treesitter_queries/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+pub mod queries;
+
+use std::slice::Iter;
+
+use queries::{Query, QueryResult};
+
+pub struct TreeSitterQueriesExecutor<'a> {
+    root_node: tree_sitter::Node<'a>,
+    stmt: &'a str,
+    results: Vec<QueryResult<'a>>,
+}
+
+impl<'a> TreeSitterQueriesExecutor<'a> {
+    pub fn new(root_node: tree_sitter::Node<'a>, stmt: &'a str) -> Self {
+        Self {
+            root_node,
+            stmt,
+            results: vec![],
+        }
+    }
+
+    #[allow(private_bounds)]
```

Comment:
Keeping the `Query` trait private makes it such that we can only ever use the function with the implementations in the `queries` mod, I thought that was a cool pattern